lib/slappy/subtype_listner.rb

```diff
@@ -0,0 +1,18 @@
+module Slappy
+  class SubtypeListener
+    def initialize(subtype, callback)
+      @subtype = subtype
+      @callback = callback
+    end
+
+    def call(event)
+      return unless time_valid?(event)
+      return unless event.subtype == @subtype
+      @callback.call(event)
+    end
+
+    def time_valid?(event)
+      Time.at(event.ts.to_f) > Slappy.client.start_time
```

Comment:
Rails/TimeZone: Do not use `Time.at` without zone. Use one of `Time.zone.at`, `Time.at.current`, `Time.at.in_time_zone`, `Time.at.utc`, `Time.at.getlocal`, `Time.at.iso8601`, `Time.at.jisx0301`, `Time.at.rfc3339`, `Time.at.to_i`, `Time.at.to_f` instead.